source/game_sa/Camera.cpp

```diff
@@ -857,15 +1170,49 @@ void CCamera::Enable1rstPersonWeaponsCamera() {
 
 // 0x50CB60
 void CCamera::SetCamCollisionVarDataSet(int32 index) {
-    plugin::Call<0x50CB60, int32>(index);
+    int8 byteIndex = static_cast<int8>(index);
+
+    if (byteIndex == gCurCamColVars) {
+        return;
+    }
+
+    gCurCamColVars = byteIndex;
+    gCurDistForCam = 0x3F800000;
+    gpCamColVars = reinterpret_cast<int32>(&gCamColVars[index * 24]);
```

Comment:
``float gCamColVars[28][6]``

```suggestion
    if (index != (uint8)gCurCamColVars) {
        gCurCamColVars = index;
        gCurDistForCam = 1.0f;
        gpCamColVars   = &gCamColVars[index];
    }
```

Reply:
It seems that someone asked me to use the inverse check, and also to convert variables to the required types

(Translated via google translator)

Reply:
I checked the data type ``gCurCamColVars'' and now the type conversion is unnecessary. (suggested code also updated)